fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/impl/networking/AbstractNetworkAddon.java

```diff
@@ -172,4 +178,21 @@ public final void handleDisconnect() {
 	 * @return whether the channel is reserved
 	 */
 	protected abstract boolean isReservedChannel(Identifier channelName);
+
+	public boolean triggerCookieFuture(Identifier cookieId, byte[] cookie) {
+		CompletableFuture<byte[]> future = pendingCookieRequests.remove(cookieId);
+		if (future == null) return false;
+		future.complete(cookie);
+		return true;
+	}
+
+	public CompletableFuture<byte[]> getCookie(ClientConnection connection, Identifier cookieId) {
```

Comment:
This should all be in `AbstractChanneledNetworkAddon`, you then have access to the connection.

Reply:
Done